File: apps/android/feature/exercises/ui/src/main/kotlin/com/scribblefit/feature/exercises/ui/ExerciseTrendsState.kt
package com.scribblefit.feature.exercises.ui

import androidx.compose.runtime.Composable
import androidx.compose.runtime.ReadOnlyComposable
import androidx.compose.ui.res.stringResource
import com.scribblefit.core.config.domain.Weight
import com.scribblefit.core.model.TrendDirection
import com.scribblefit.feature.exercises.domain.usecase.TrendDataPoint
import com.scribblefit.feature.exercises.domain.usecase.TrendInsights
import com.scribblefit.feature.exercises.domain.usecase.TrendPeriod

data class ExerciseTrendsState(
    val exerciseName: String = "",
    val isLoading: Boolean = false,
    
    // OneRM Data
    val oneRMDataPoints: List<TrendDataPoint> = emptyList(),
    val oneRMInsights: TrendInsights? = null,
    
    // Volume Data
    val volumeDataPoints: List<TrendDataPoint> = emptyList(),
    val volumeInsights: TrendInsights? = null,
    
    val weightUnit: Weight = Weight.LBS,
    val selectedPeriod: TrendPeriod = TrendPeriod.THREE_MONTHS,
    val error: String? = null
) {
    val navigationTitle: String get() = exerciseName
    
    val oneRMSectionTitle: String
        @Composable @ReadOnlyComposable
        get() = stringResource(R.string.exercise_trends_estimated_1rm)
        
    val volumeSectionTitle: String
        @Composable @ReadOnlyComposable
        get() = stringResource(R.string.exercise_trends_total_volume)
        
    val emptyDataMessage: String
        @Composable @ReadOnlyComposable
        get() = stringResource(R.string.exercise_trends_no_data)
        
    val periodPickerLabel: String
        @Composable @ReadOnlyComposable
        get() = stringResource(R.string.exercise_trends_period_label)
        
    val personalBestLabel: String
        @Composable @ReadOnlyComposable
        get() = stringResource(R.string.exercise_trends_pb_label)
        
    val weightUnitLabel: String
        @Composable @ReadOnlyComposable
        get() = if (weightUnit == Weight.KGS) {
            stringResource(R.string.workout_exercises_weight_unit_kg)
        } else {
            stringResource(R.string.workout_exercises_weight_unit_lb)
        }
        
    @Composable
    @ReadOnlyComposable
    fun getPeriodText(period: TrendPeriod): String =
        when (period) {
            TrendPeriod.ONE_MONTH -> stringResource(R.string.exercise_trends_period_1m)
            TrendPeriod.THREE_MONTHS -> stringResource(R.string.exercise_trends_period_3m)
            TrendPeriod.SIX_MONTHS -> stringResource(R.string.exercise_trends_period_6m)
            TrendPeriod.ONE_YEAR -> stringResource(R.string.exercise_trends_period_1y)
            TrendPeriod.ALL -> stringResource(R.string.exercise_trends_period_all)
        }

    @Composable
    @ReadOnlyComposable
    fun getTrendBadgeText(direction: TrendDirection, percentage: Float): String =
        when (direction) {
            TrendDirection.IMPROVING -> "+${percentage.toInt()}%"
            TrendDirection.STABLE -> stringResource(R.string.trend_stable)
            TrendDirection.PLATEAUED -> stringResource(R.string.trend_plateaued)
            TrendDirection.DECLINING -> "${percentage.toInt()}%"
        }
}

File: apps/ios/ScribbleFit/LocalPackages/Features/FeatureExercises/Sources/FeatureExercises/UI/ExerciseTrendsState.swift
import Foundation
import CoreModel

public struct ExerciseTrendsState: Sendable {
    public var exerciseName: String
    public var isLoading: Bool = false

    // OneRM Data
    public var oneRMDataPoints: [TrendDataPoint] = []
    public var oneRMInsights: TrendInsights?

    // Volume Data
    public var volumeDataPoints: [TrendDataPoint] = []
    public var volumeInsights: TrendInsights?

    public var error: String?
    public var weightUnit: WeightUnit = .lbs
    public var selectedPeriod: TrendPeriod = .threeMonths

    public init(exerciseName: String) {
        self.exerciseName = exerciseName
    }

    // Resolved Strings
    public var navigationTitle: String { exerciseName }
    public var oneRMSectionTitle: String { String(localized: "Estimated 1RM") }
    public var volumeSectionTitle: String { String(localized: "Total Volume") }
    public var emptyDataMessage: String { String(localized: "Not enough data for trends yet") }
    public var periodPickerLabel: String { String(localized: "Period") }
    public var personalBestLabel: String { String(localized: "PB") }

    public func badgeText(for direction: TrendDirection, percentage: Float) -> String {
        switch direction {
        case .improving: return "+\(Int(percentage))%"
        case .stable: return String(localized: "STABLE")
        case .plateaued: return String(localized: "PLATEAUED")
        case .declining: return "\(Int(percentage))%"
        }
    }

    public var weightUnitLabel: String {
        weightUnit == .kgs ? String(localized: "kg") : String(localized: "lbs")
    }
}
